src/responder/common/cache_req/cache_req.c

```diff
@@ -971,8 +991,10 @@ static void cache_req_search_domains_done(struct tevent_req *subreq)
     ret = cache_req_search_recv(state, subreq, &result, &dp_success);
     talloc_zfree(subreq);
 
-    /* Remember if any DP request fails. */
-    state->dp_success = !dp_success ? false : state->dp_success;
+    /* Remember if any DP request fails, if DP was contacted. */
+    if (cache_req_dp_contacted(state)) {
+        state->dp_success = !dp_success ? false : state->dp_success;
+    }
```

Comment:
What value `dp_success` gets in `else` case?
It was set to `true` by default when request state was create?

Why `cache_req_dp_contacted()` is required - wouldn't it be possible to deduce all required information from `dp_success` (perhaps by changing init value)?

Reply:
It is set to true by default. And it is set to false if any dp request fails. Now, if we do not contact dp, it is unchanged. Previously it was set to false.

I guess we can initialize it to true in cache_req_search_send (state->dp_success) so if no dp was contacted, true is returned from cache_req_search_recv. But current change feels more clear to me.

Reply:
My worry is that is adds yet another one place with a switch over `cache_behavior`. This means another place to not miss if a new 'case' every introduced. But ok.